blocks/bricks/recurrent.py

```diff
@@ -772,9 +771,9 @@ def suffixes(names, level):
     def split_suffix(name):
         # Target name with suffix to the correct layer
         name_level = name.split('_')
-        if len(name_level) == 2:
-            name, level = name_level
-            level = int(level)
+        if name_level[-1].isdigit():
```

Comment:
This will also fail if the states of the core transition have names ending with digits.

I would suggest to use a more exotic separator, e.g. `$` or `#` to indicate the name mangled states.